### PR TITLE
feat: allow store variants under step template

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

Update the POST /workflows endpoint to support the variants on the steps.
The parts of the code we have to update:
variables for all steps + variants
create message templates for variants
changes for variants

## Definition of Done
the POST /workflows endpoint supports the variants on the steps
e2e tests

### CODE REVIEW

File: apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts [hunk 1]
```diff
@@ -79,97 +100,185 @@ export class CreateNotificationTemplate {
...
+          actor: variant.template.actor,
+          parentChangeId,
+        })
+      );
+
+      variantsList.push({
+        _id: variantTemplate._id,
+        _templateId: variantTemplate._id,
+        filters: variant.filters,
+        _parentId: parentStepId,
```
[the remark this answers is not in the document]
Comment (djabarovgeorge) [reply apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts:267]: Totally right I mistyped it. Thanks!!

File: apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts [hunk 2]
```diff
@@ -79,97 +100,185 @@ export class CreateNotificationTemplate {
...
           layoutId: message.template.layoutId,
           preheader: message.template.preheader,
           senderName: message.template.senderName,
-          parentChangeId,
           actor: message.template.actor,
+          parentChangeId,
         })
       );
 
+      const storedVariants = await this.storeVariantSteps(message, command, parentChangeId, parentStepId);
```
[the remark this answers is not in the document]
Comment (djabarovgeorge) [reply apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts:197]: Definitely could be ;)

File: apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts [hunk 3]
```diff
@@ -79,97 +100,185 @@ export class CreateNotificationTemplate {
...
+    );
+  }
+
+  private async storeWorkflow(
+    command: CreateNotificationTemplateCommand,
+    templateSteps: INotificationTemplateStep[],
+    trigger: INotificationTrigger,
+    triggerIdentifier: string
+  ) {
+    const tmp = {
```
[the remark this answers is not in the document]
Comment (djabarovgeorge) [reply apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts:137]: oh my, i made it while making a refactor but in the end, i reverted it but the name remained 🤦‍♂️

File: apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts [hunk 4]
```diff
@@ -37,15 +43,30 @@ export class CreateNotificationTemplate {
...
-
     const triggerIdentifier = `${slugify(command.name, {
       lower: true,
       strict: true,
     })}`;
+    const parentChangeId: string = NotificationTemplateRepository.createObjectId();
+
+    const templateSteps = await this.storeTemplateSteps(command, parentChangeId);
+
+    const trigger = await this.createNotificationTrigger(command, triggerIdentifier);
```
[the remark this answers is not in the document]
Comment (djabarovgeorge) [reply apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts:54]: definitely ;)

File: apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts [hunk 5]
```diff
@@ -37,15 +43,31 @@ export class CreateNotificationTemplate {
...
     const triggerIdentifier = `${slugify(command.name, {
       lower: true,
       strict: true,
     })}`;
+    const parentChangeId: string = NotificationTemplateRepository.createObjectId();
+
+    const [templateSteps, trigger] = await Promise.all([
+      await this.storeTemplateSteps(command, parentChangeId),
+      await this.createNotificationTrigger(command, triggerIdentifier),
+    ]);
```
Comment (ainouzgali) [inline apps/api/src/app/workflows/usecases/create-notification-template/create-notification-template.usecase.ts:55]:
```suggestion
    const [templateSteps, trigger] = await Promise.all([
      this.storeTemplateSteps(command, parentChangeId),
      this.createNotificationTrigger(command, triggerIdentifier),
    ]);
```

**Review by ainouzgali — APPROVED** [review]

**Pull request merged**